### PR TITLE
Create a spider generator

### PR DESCRIPTION
**Pull request by anapaulagomes**
This script is for cases where we have a few spiders that are not complex and from the same source.

Example of a CSV file:
```
url,city,state,territory_id,start_day,start_month,start_year,base_class
http://pmXIQUEXIQUEBA.imprensaoficial.org,Xique-Xique,BA,2933604,1,1,2017,ImprensaOficialSpider
http://pmWENCESLAUGUIMARAESBA.imprensaoficial.org,Wenceslau Guimarães,BA,2933505,1,1,2017,ImprensaOficialSpider
http://pmVERACRUZBA.imprensaoficial.org,Vera Cruz,BA,2933208,1,4,2017,ImprensaOficialSpider
```

Command:
```
python generate_spiders.py new-spiders.csv
```

Example of a generated spider:

```
from datetime import date
from gazette.spiders.base import ImprensaOficialSpider


class BaGentioDoOuroSpider(ImprensaOficialSpider):

    name = "ba_gentio_do_ouro"
    allowed_domains = ["pmGENTIODOOUROBA.imprensaoficial.org"]
    start_date = date(2017, 2, 1)
    url_base = "http://pmGENTIODOOUROBA.imprensaoficial.org"
    TERRITORY_ID = "2911303"
```

### CODE REVIEW

File: scripts/generate_spiders.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            spider_data = {
+                "spider_class_base": row["base"],
+                "spider_class_name": f"{row['state'].title()}{row['city'].title()}",
+                "spider_name": f"{row['state'].lower()}_{row['city'].lower()}",
+                "allowed_domain": row["domain"],
+                "start_date": "",
+                "base_url": row["base_url"],
+                "territory_id": row["territory_id"],
+            }
+            load_template(spider_data)
```
Comment (giuliocc) [inline scripts/generate_spiders.py:39]: The suggestions are simple, but as it changes most parts of the script, it became a suggestion for the whole file :grimacing:

Basically, I suggest a way to deal with multi-word cities, non-ascii characters and also export the content to the respective files.
```suggestion
import argparse
import csv
from os.path import abspath, dirname
from unicodedata import normalize

from jinja2 import Environment, FileSystemLoader


def load_template(spider_data):
    env = Environment(loader=FileSystemLoader("."))
    template = env.get_template(f"spider_template.py")
    return template.render(
        spider_class_base=spider_data["spider_class_base"],
        spider_class_name=spider_data["spider_class_name"],
        spider_name=spider_data["spider_name"],
        allowed_domain=spider_data["allowed_domain"],
        start_date=spider_data["start_date"],
        base_url=spider_data["base_url"],
        territory_id=spider_data["territory_id"],
    )


if __name__ == "__main__":
    parser = argparse.ArgumentParser("Generate a spider structure from CSV.")
    parser.add_argument("file")
    args = parser.parse_args()

    project_root = dirname(dirname(abspath(__file__)))

    with open(args.file) as csv_file:
        reader = csv.DictReader(csv_file)
        for row in reader:
        	normalized_city_name = (
                normalize("NFKD", row["city"]).encode("ASCII", "ignore").decode("utf8")
            )
            split_city_name = normalized_city_name.replace("-", " ").split()

            spider_data = {
                "spider_class_base": row["base"],
                "spider_class_name": f"{row['state'].title()}{''.join(split_city_name).title()}",
                "spider_name": f"{row['state'].lower()}_{'_'.join(split_city_name).lower()}",
                "allowed_domain": row["domain"],
                "start_date": "",
                "base_url": row["base_url"],
                "territory_id": row["territory_id"],
            }

            filepath = f"{project_root}/data_collection/gazette/spiders/{row['state'].lower()}_{'_'.join(split_city_name).lower()}.py"

            with open(filepath, "w") as spiderfile:
                spiderfile.write(load_template(spider_data))
```

Comment (anapaulagomes) [reply scripts/generate_spiders.py]: Thanks for your suggestions, @giuliocc! I've implemented most of it. :) Works like a charm.

**Comment by jvanz** [comment]
Is this a draft or is open to review?

**Comment by anapaulagomes** [comment]
Hi @jvanz, I was waiting for the Travis checks before marking it as ready. It's ready to be reviewed.

**Review by jvanz — COMMENTED** [review]
LGTM. But I would suggest add documentation about how use this script. I think a simple example, should be enough.

File: README.md [hunk 2]
```diff
@@ -69,6 +70,47 @@ following command will download only gazettes which date is greater than 01/Sep/
...
+Once you have the CSV file, run the command:
+
+```
+cd scripts/
+
+python generate_spiders.py new-spiders.csv
+```
+
+That's it. The new spiders will be in the directory `data_collection/gazette/spiders/`.
+
```
Comment (jvanz) [inline README.md:113]: :heart:

**Review by jvanz — APPROVED** [review]

**Comment by jvanz** [comment]
Sorry for long time with no response. =/

**Pull request merged**